pkg/reconciler/taskrun/taskrun.go

```diff
@@ -269,9 +269,30 @@ func (c *Reconciler) prepare(ctx context.Context, tr *v1beta1.TaskRun) (*v1beta1
 	// and may not have had all of the assumed default specified.
 	tr.SetDefaults(contexts.WithUpgradeViaDefaulting(ctx))
 
-	getTaskfunc, err := resources.GetTaskFuncFromTaskRun(ctx, c.KubeClientSet, c.PipelineClientSet, tr)
+	var (
+		err      error
+		taskMeta metav1.ObjectMeta
+		taskSpec v1beta1.TaskSpec
+	)
+
+	if tr.Status.TaskSpec == nil {
+		err = fmt.Errorf("task spec expected to be cached in status of taskrun %s/%s but it was not found", tr.Namespace, tr.Name)
```

Comment:
TODO: update message to include mention that task spec is expected to be cached _by a resolver_.